api/v1alpha1/connection_types.go

```diff
@@ -5,14 +5,25 @@
 
 package v1alpha1
 
-import gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
 
 // Connection allows users to configure connection-level settings
 type Connection struct {
 	// ConnectionLimit defines limits related to connections
 	//
 	// +optional
 	ConnectionLimit *ConnectionLimit `json:"connectionLimit,omitempty"`
+	// BufferLimit provides configuration for the maximum buffer size in bytes for each incoming connection.
+	// For example, 20Mi, 1Gi, 256Ki etc.
+	// Note that when the suffix is not provided, the value is interpreted as bytes.
+	// Default: 32768 bytes.
+	//
+	// +kubebuilder:validation:XValidation:rule="self.matches(r\"^[1-9]+[0-9]*([EPTGMK]i|[EPTGMk])?$\")",message="bufferLimit must be of the format \"^[1-9]+[0-9]*([EPTGMK]i|[EPTGMk])?$\""
```

Comment:
I suggest changing the CEL rule to:

`type(self) == string ? self.matches(r"^[1-9]+[0-9]*([EPTGMK]i|[EPTGMk])?$") : true`

This way it works for raw numbers as well as for string values.

[CEL playground](https://playcel.undistro.io/?content=H4sIAAAAAAAAAyWMTwuCQBTEv8rj1UH7I3rokCDSIYrKCuoSuwaSb3OpVtldoai%2BeytdZn7MDPPGsrAFxtiDJWkCacBWzlTTWugqh3CaZRuoNawOuy2IWj8KG3DFlaG7iCGahKEoRYkjpGejyRhZK%2FdoXw153cSHJAFjtVRXSKFLAvdwqch4muOZReNpPmSh04HH5vvjIlvn8vOnW%2B6nfY4%2BxGB1S%2Fj9AZEILbWxAAAA)

Reply:
Thanks, that did the trick